app/src/main/java/com/sopt/now/compose/ui/screen/SignupScreen.kt

```diff
@@ -0,0 +1,207 @@
+                userPw = newValue }, // 이 부분 안쓰면 입력안됨
+            modifier = modifier
+                .fillMaxWidth(),
+            label = { Text("비밀번호를 입력해주세요") },
+            placeholder = { Text("123456") },
+            singleLine = true,
+        )
+
+        // 높이 30dp인 공간 추가
+        Spacer(modifier = modifier.height(30.dp))
+
+        // nickname
+        Text(
+            text = "닉네임",
+
+            fontSize = 20.sp,
+            modifier = modifier.fillMaxWidth(),
+            textAlign = TextAlign.Left
+        )
+
+        TextField(
+            value = userName,
+            onValueChange = { newValue ->
+                userName = newValue }, // 이 부분 안쓰면 입력안됨
+            modifier = modifier
+                .fillMaxWidth(),
+            label = { Text("닉네임 입력해주세요") },
+            placeholder = { Text("Hyunjin Lee") },
+            singleLine = true,
+        )
+
+        // 높이 30dp인 공간 추가
+        Spacer(modifier = modifier.height(30.dp))
+
+        // nickname
+        Text(
+            text = "MBTI",
+
+            fontSize = 20.sp,
+            modifier = modifier.fillMaxWidth(),
+            textAlign = TextAlign.Left
+        )
+
+        TextField(
+            value = userMbti,
+            onValueChange = { newValue ->
+                userMbti = newValue }, // 이 부분 안쓰면 입력안됨
+            modifier = modifier
+                .fillMaxWidth(),
+            label = { Text("닉네임 입력해주세요") },
+            placeholder = { Text("entj") },
+            singleLine = true,
+        )
+
+        // 2 가중치 만큼 빈공간 추가
+        Spacer(modifier = modifier.weight(2f))
+
+        // signup button
+        val context = LocalContext.current
+        Button(
+            onClick = {
+                if (CheckId(userId) && CheckPw(userPw) && CheckName(userName) && CheckMbti(userMbti)) {
+                    // ViewModel에 사용자 정보 저장
+                    userViewModel.userId.value = userId
+                    userViewModel.userPw.value = userPw
+                    userViewModel.userName.value = userName
+                    userViewModel.userMbti.value = userMbti
+
+                    Toast.makeText(context, "회원가입 성공", Toast.LENGTH_SHORT).show()
+                    navController.navigate("login")
+                }
+            },
+            modifier = modifier
+                .padding(10.dp)
+                .fillMaxWidth(),
+            shape = RoundedCornerShape(30.dp)
+        ) {
+            Text("회원가입하기", color = Color.White)
+        }
+    }
+}
+
+fun CheckId(id: String): Boolean {
+    if(id.length >= 6 && id.length <= 10){
+        return true
+    }
+    else return false
+}
+
+fun CheckPw(pw: String): Boolean {
+    if(pw.length >= 8 && pw.length <= 12){
+        return true
+    }
+    else return false
+}
+
+fun CheckName(name: String): Boolean {
+    if(name.length == 1 && name == " "){
+        return false
+    }
+    else if(name.length >= 1) return true
+    else return false
+}
```

Comment:
```suggestion

fun CheckName(name: String): Boolean  = when {
     name.length == 1 && name == " " -> false
    else -> name.isNotEmpty()
}
```
확실하진 않지만,, 이런 식으로 when문으로 처리해주면 더 가독성 있을 것 같습니다!
그리고 `name.length`가 1인 조건이 겹치는 것 같네요!